prow/plugins/milestoneapplier/milestoneapplier.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package milestoneapplier implements the plugin to automatically apply
+// the configured milestone after a PR is merged.
+package milestoneapplier
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/plugins/milestone"
+)
+
+const pluginName = "milestoneapplier"
+
+type githubClient interface {
+	SetMilestone(org, repo string, issueNum, milestoneNum int) error
+	ListMilestones(org, repo string) ([]github.Milestone, error)
+}
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	configInfo := map[string]string{}
+	for _, orgRepo := range enabledRepos {
+		configInfo[orgRepo] = fmt.Sprintf("The current milestone for this repo is %s.", config.MilestoneApplier[orgRepo])
```

Comment:
Wouldn't this vary by branch?

Reply:
Also, do you want a structure that allows you to specify a long list of repos where the branch --> milestone mapping is identical (e.g. how the Bugzilla config works)?